server/src/main/java/org/opensearch/identity/MyShiroModule.java

```diff
@@ -0,0 +1,118 @@
+     */
+    public static Subject getSubjectOrInternal() {
+        final Subject existingSubject = SecurityUtils.getSubject();
+        if (existingSubject.isAuthenticated()) {
+            return existingSubject;
+        }
+
+        // Used for tracing where this function is used that is visible in the log output
+        final Exception e = new Exception();
+        final StackTraceElement current = e.getStackTrace()[1];
+        final String sourceAnnotation = current.getFileName() + "." + current.getMethodName() + "@" + current.getLineNumber();
+
+        final Subject internalSubject = new Subject.Builder().authenticated(true)
+            .principals(new SimplePrincipalCollection("INTERNAL-" + sourceAnnotation, "OpenSearch")) // How can we ensure the roles this
+                                                                                                     // princpal resolves?
+            .contextAttribute("NodeId", "???") // Can we use this to source the originating node in a cluster?
+            .buildSubject();
+        return internalSubject;
+    }
+
+    /**
+     * Attempt to authenticate via authorization header, ignores if already authenticated, throws exceptions if unable
+     *
+     * NOTE: this was quickly built for test scenarios and will not be used log term, there is a supplimental library for
+     * Shiro that supports web based applications, including https://shiro.apache.org/static/1.9.1/apidocs/org/apache/shiro/web/filter/authc/BasicHttpAuthenticationFilter.html
+     * if this can be reused and to what degree will be useful in future investigations.
+     */
+    public static void authenticateViaAuthorizationHeader(final Optional<String> authHeader) throws AuthenticationException {
+        final Subject currentSubject = SecurityUtils.getSubject();
+        if (currentSubject.isAuthenticated()) {
+            // No need to authenticate already ready
+            return;
+        }
+
+        if (authHeader.isPresent() && !(Strings.isNullOrEmpty(authHeader.get())) && authHeader.get().startsWith("Basic ")) {
+            final byte[] decodedAuthHeader = Base64.getDecoder().decode(authHeader.get().substring(6));
+            final String[] decodedUserNamePassword = new String(decodedAuthHeader).split(":");
+            currentSubject.login(new UsernamePasswordToken(decodedUserNamePassword[0], decodedUserNamePassword[1]));
+            // Successful login - return!
+            return;
+        }
+
+        throw new AuthenticationException("Unable to authenticate user!");
+    }
+
+    /* Super basic role management */
+    private enum Roles {
+        ALL_ACCESS,
+        CLUSTER_MONITOR;
+    }
+
+    /** Very basic user pool and permissions ecosystem */
+    private static class MyRealm extends SimpleAccountRealm {
+        private MyRealm() {
+            super("OpenSearch");
+
+            /* Default account configuration */
+            this.addAccount("admin", "admin", Roles.ALL_ACCESS.name());
```

Comment:
I know this is just a prototype, but adding hardcoded identities like this seems really sketchy.  Can we pull the "addUser" stuff out of code and into some sort of config script or command line arguments?